src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -68,107 +68,199 @@ def validate_location(namespace):
         else:
-            # last try: is it a custom image name?
-            compute_client = _compute_client_factory()
-            try:
-                compute_client.images.get(namespace.resource_group_name, image)
-                image = namespace.image = _get_resource_id(image, namespace.resource_group_name,
-                                                           'images', 'Microsoft.Compute')
-                namespace.storage_profile = StorageProfile.ManagedCustomImage
-            except CloudError:
-                err = 'Invalid image "{}". Use a custom image name, id, or pick one from {}'
-                raise CLIError(err.format(image, [x['urnAlias'] for x in images]))
+            # STORAGE PROFILE #6
+            namespace.storage_profile = StorageProfile.ManagedSpecializedOSDisk
+    elif namespace.image and not getattr(namespace, 'attach_os_disk', None):
+        image_type = _parse_image_argument(namespace)
+        if image_type == 'uri':
+            # STORAGE PROFILE #2
+            namespace.storage_profile = StorageProfile.SACustomImage
+        elif image_type == 'image_id':
+            # STORAGE PROFILE #5
+            namespace.storage_profile = StorageProfile.ManagedCustomImage
+        elif image_type == 'urn':
+            if namespace.use_unmanaged_disk:
+                # STORAGE PROFILE #1
+                namespace.storage_profile = StorageProfile.SAPirImage
+            else:
+                # STORAGE PROFILE #4
+                namespace.storage_profile = StorageProfile.ManagedPirImage
+        else:
+            raise CLIError('Unrecognized image type: {}'.format(image_type))
+    else:
+        # did not specify image XOR attach-os-disk
+        raise CLIError('incorrect usage: --image IMAGE | --attach-os-disk DISK')
+
+    # perform parameter validation for the specific storage profile
+    # start with the required/forbidden parameters for VM
+    if namespace.storage_profile == StorageProfile.ManagedPirImage:
+        required = ['image']
+        forbidden = ['os_type', 'attach_os_disk', 'storage_account',
+                     'storage_container_name', 'use_unmanaged_disk']
+        if for_scale_set:
+            forbidden.append('os_disk_name')
+        _validate_managed_disk_sku(namespace.storage_sku)
+
+    elif namespace.storage_profile == StorageProfile.ManagedCustomImage:
+        required = ['image']
+        forbidden = ['os_type', 'attach_os_disk', 'storage_account',
+                     'storage_container_name', 'use_unmanaged_disk']
+        if for_scale_set:
+            forbidden.append('os_disk_name')
+        _validate_managed_disk_sku(namespace.storage_sku)
+
+    elif namespace.storage_profile == StorageProfile.ManagedSpecializedOSDisk:
+        required = ['os_type', 'attach_os_disk']
+        forbidden = ['os_disk_name', 'storage_caching', 'storage_account',
+                     'storage_container_name', 'use_unmanaged_disk']
```

Comment:
i think `storage_sku` should be forbidden as well, same for the `SASpecializedOSDisk` option below